ui/packages/catalog-ui-search/src/main/webapp/js/router.js

```diff
@@ -11,298 +11,106 @@
-                    this.updateRoute(name, path, args);
-                    break;
-                case 'home':
-                    if (Application.App.workspacesRegion.currentView===undefined) {
-                        Application.App.workspacesRegion.show(new HomeView());
-                    }
-                    Application.App.workspacesRegion.$el.removeClass('is-hidden');
-                    this.updateRoute(name, path, args);
-                    break;
-                case 'workspaces':
-                    if (Application.App.workspacesRegion.currentView===undefined) {
-                        Application.App.workspacesRegion.show(new HomeView());
-                    }
-                    Application.App.workspacesRegion.$el.removeClass('is-hidden');
-                    this.updateRoute(name, path, args);
-                    break;
-                case 'openSources':
-                    if (Application.App.sourcesRegion.currentView === undefined) {
-                        Application.App.sourcesRegion.show(new SourcesView());
-                    }
-                    Application.App.sourcesRegion.$el.removeClass('is-hidden');
-                    this.updateRoute(name, path, args);
-                    break;
-                case 'openAbout':
-                    if (Application.App.aboutRegion.currentView === undefined) {
-                        Application.App.aboutRegion.show(new AboutView());
-                    }
-                    Application.App.aboutRegion.$el.removeClass('is-hidden');
-                    this.updateRoute(name, path, args);
-                    break;
-                case 'notFound':
-                    if (Application.App.notFoundRegion.currentview === undefined) {
-                        Application.App.notFoundRegion.show(new NotFoundView());
-                    } 
-                    Application.App.notFoundRegion.$el.removeClass('is-hidden');
-                    this.updateRoute(name, path, args);
-                    break;
-            }
+        openUpload: {
+            patterns: ['uploads/:id'],
+            component: require('component/upload/upload.view')
         },
-        updateRoute: function(name, path, args){
+        openSources: {
+            patterns: ['sources(/)'],
+            component: require('component/sources/sources.view')
+        },
+        openAbout: {
+            patterns: ['about(/)'],
+            component: require('component/about/about.view')
+        }
+    }),
+    // needs to be last based on how backbone router works, otherwise this route always wins
+    notFound: {
+        patterns: ['*path'],
+        component: require('component/notfound/notfound.view')
+    }    
+};
+
+initializeRoutes(routeDefinitions);
+
+const Router = Backbone.Router.extend({
+    routes: Object.keys(routeDefinitions).reduce((routesBlob, key) => {
+        const { patterns } = routeDefinitions[key]
```

Comment:
Instead of obtaining the key from `routeDefinitions`, then going back to `routeDefinitions` again with the key to get the value, `Object.entries()` could be used to get the key-value pairs. For code length concerns, `key` and `patterns` could be obtained in the lambda signature using the destructuring assignment `[key, { patterns }]`